python_modules/dagster/docs/reference.rst

```diff
@@ -0,0 +1,212 @@
+class. We also provide a `thin wrapper <https://github.com/dagster-io/dagster/tree/master/python_modules/libraries/dagster-ge>`_
+around the `great_expectations <https://github.com/great-expectations/great_expectations>`_ library
+so you can use its existing repertoire of expectartions with Dagster.
+
+.. _pipeline:
+
+Pipeline
+^^^^^^^^
+Data pipelines are directed acyclic graphs (DAGs) of solids -- that is, they are made up of a number
+of solids which have data `dependencies <#dependency-definition>`__ on each other (but no circular
+dependencies), along with a set of associated pipeline context definitions, which declare the various
+environments in which a pipeline can execute.
+
+Pipelines are defined using the :class:`PipelineDefinition <dagster.PipelineDefinition>` class, and
+their contexts are defined using :class:`PipelineContextDefinition <dagster.PipelineContextDefinition>`.
+
+When a pipeline is combined with a given config conforming to one of its declared contexts, it can
+be compiled by the Dagster engine into an execution plan that can be executed on various compute
+substrates.
+
+Concretely, a pipeline might include context definitions for local testing (where databases and
+other resources will be mocked, in-memory, or local) and for running in production (where resources
+will require different credentials and expose configuration options). When a pipeline is compiled
+with a config corresponding to one of these contexts, it yields an execution plan suitable for the
+given environment.
+
+Repository
+^^^^^^^^^^
+
+A repository is a collection of pipelines that can be made available to the Dagit UI and other
+higher-level tools. Repositories are defined using the
+:class:`RepositoryDefinition <dagster.RepositoryDefinition>` class, and made available to
+higher-level tools with a special ``repository.yml`` file that tells the tools where to look for a
+repository definition.
+
+Config
+^^^^^^
+
+Config defines the external environment with which a pipeline will interact for a given execution
+plan. Config can be used to change solid behavior, define pipeline- or solid-scoped resources and
+data that will be available during execution, or even shim solid inputs.
+
+Config is complementary to data (solid inputs and outputs) -- think of inputs and outputs as
+specifying `what` data a pipeline operates on, and config as specifying `how` it operates.
+
+Concretely, imagine a pipeline of solids operating on a data warehouse. The solids might emit and
+consume table partition IDs and aggregate statistics as inputs and outputs -- the data on which they
+operate. Config might specify how to connect to the warehouse (so that the pipeline could also
+operate against a local test database), how to log the results of intermediate computations, or
+where to put artifacts like plots and summary tables.
+
+Config Fields
+^^^^^^^^^^^^^
+
+Config fields define a schema for how users can config pipelines (using either Python dicts, YAML,
```

Comment:
I think we should avoid using the word "config" too much given its potential aliasing issues. 

For example we could call this section `Field`s or `Configuration Schema`s and when used as a verb say configure.

Reply:
great, preference between Field and Configuration Schema?

Reply:
I would go Configuration Schema and then mention within `Field`, `Dict` and whatever else are "Dagster Type"s that can only be used in the configuration schema.